fix: addressing feedback from testing

mikejgray opened the pull request.

mikejgray committed:
fix: addressing feedback from testing

mikejgray committed:
no one likes commented code anyway

NeonDaniel commented:
Minor nits:
- `installer.sh` is not marked executable
- The "Neon Hub Installer" screen still prints `Neon.AI` instead of `Neon AI ®`

NeonDaniel commented:
![Truncated Hub Installer](https://github.com/user-attachments/assets/175ebbc6-ffb5-4fbe-8e47-03a9f3d96004)

NeonDaniel commented:
I see the same errors after re-running the installer. I just pulled the `dev` branch and re-ran `setup.sh`; let me know if you'd like me to do anything differently
```
BOOT FAILED
2024-10-16 19:09:27.391063+00:00 [error] <0.253.0>
2024-10-16 19:09:27.391063+00:00 [error] <0.253.0> BOOT FAILED
2024-10-16 19:09:27.391063+00:00 [error] <0.253.0> ===========
2024-10-16 19:09:27.391063+00:00 [error] <0.253.0> Error during startup: {error,failed_to_update_enabled_plugins_file}
2024-10-16 19:09:27.391063+00:00 [error] <0.253.0>
===========
Error during startup: {error,failed_to_update_enabled_plugins_file}

2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>   crasher:
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>     initial call: application_master:init/4
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>     pid: <0.252.0>
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>     registered_name: []
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>     exception exit: {failed_to_update_enabled_plugins_file,
Runtime terminating during boot (terminating)
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>                         {rabbit,start,[normal,[]]}}
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>       in function  application_master:init/4 (application_master.erl, line 142)
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>     ancestors: [<0.251.0>]
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>     message_queue_len: 1

2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>     messages: [{'EXIT',<0.253.0>,normal}]
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>     links: [<0.251.0>,<0.44.0>]
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>     dictionary: []
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>     trap_exit: true
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>     status: running
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>     heap_size: 376
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>     stack_size: 28
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>     reductions: 167
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>   neighbours:
2024-10-16 19:09:28.391574+00:00 [error] <0.252.0>
2024-10-16 19:09:28.394456+00:00 [notice] <0.44.0> Application rabbit exited with reason: {failed_to_update_enabled_plugins_file,{rabbit,start,[normal,[]]}}
Crash dump is being written to: erl_crash.dump...[os_mon] cpu supervisor port (cpu_sup): Erlang has closed
[os_mon] memory supervisor port (memsup): Erlang has closed
```

mikejgray committed:
fix: enabled_plugins, requested changes

mikejgray committed:
typo

mikejgray committed:
correction

mikejgray committed:
adjust permissions

mikejgray committed:
rmq permissions

mikejgray committed:
yacht tweaks

mikejgray committed:
bump version, better docs, better error handling

mikejgray commented:
I was able to get this version working, please give it another shot and let me know if anything else comes up.

NeonDaniel commented:
Working for me now :tada: .

Summarizing discussion from Matrix chat:
- yacht container had to be manually removed before installer would work due to name change
- some of my reported errors may be related to my not properly pulling the latest commit from the `dev` branch
- Installer is failing because `DISTRO_NAME` is unset when `required_packages` is run in the current state
- I validated functionality by browsing to the HANA endpoint and running the example calls in `get_stt`, `get_tts`, and `get_response`

Overall seems to be working for me with the exception of the distro check that I manually overrode. Updates are not part of this initial SoW, so the container update issue can be pushed to later since I think that is a broader question about how to backup/cleanup an existing installation before an update.

mikejgray committed:
preserve scope

mikejgray committed:
additional troubleshooting step

mikejgray commented:
@NeonDaniel I believe I've fixed the bug introduced in the commit you tried and added a troubleshooting step for clearing out old containers. Would you please re-validate just to be sure? It's working on my end.

mikejgray committed:
feat: detailed docs

mikejgray committed:
more tweaks

mikejgray committed:
play with newt another day

mikejgray committed:
more documentation tweaks

mikejgray committed:
more links

mikejgray committed:
a script to validate HANA/core functionality

NeonDaniel commented on docs/configuration.md:
```diff
@@ -0,0 +1,114 @@
+!!! warning
+    YAML is a whitespace-sensitive language. Be sure to maintain proper indentation when editing the `neon.yaml` file.
+
+Specific configuration changes are also detailed in blog posts by community dev Mike Gray. [Read the blog posts](https://blog.graywind.org/tags/neon/).
```
Consider also a link to [Neon docs](https://neongeckocom.github.io/neon-docs/quick_reference/configuration/)

NeonDaniel commented on docs/configuration.md:
````diff
@@ -0,0 +1,114 @@
+
+### Units of measurement
+
+To switch between metric and imperial units, change the `system_unit` value. The default value is `imperial`.
+
+```yaml
+system_unit: metric
+```
+
+To switch between 12-hour and 24-hour time, change the `time_format` value. The default value is `half`, or 12-hour time.
+
+```yaml
+time_format: full
+```
+
+### Timezone
+
+To change the timezone manually, change the `location.timezone` value. The default value is `America/Los_Angeles`.
+
+```yaml
+location:
+  timezone: America/Chicago
+```
````
This will affect the system defaults, but its worth noting that clients/nodes are generally expected to supply this information with each request and their configuration may be formatted differently

NeonDaniel commented on docs/configuration.md:
````diff
@@ -0,0 +1,114 @@
+The default `neon.yaml` file includes a commented section for using your own API keys for external services. It looks like this:
+
+```yaml
+# keys:
````
The `keys` top-level key is no longer required (but is still supported) as of [this PR](https://github.com/NeonGeckoCom/neon_api_proxy/pull/99) (v 0.7.0)

NeonDaniel reviewed (approved):
Left a few notes on documentation. We may want to move/duplicate that to the `neon-docs` repository, but this is a good place for it to live for now and it looks well-organized and complete to me!

mikejgray committed:
addressing feedback
